### PR TITLE
Enable edit and delete info-comments

### LINKED ISSUE #5702: It should be possible to edit/delete comments
**Is your feature request related to a problem? Please describe.**
To the best of my knowledge, it is currently not possible to delete or edit the content of existing "non-correction" comments. 

**Describe the solution you'd like**
It should be possible to edit and delete "non-correction" existing "non-correction" comments. 

**Additional context**
Since the effects of "correction comments" on workflow etc are more complex, the possibility of editing/deleting them should be handled separately.

### PR DESCRIPTION
**Pull request by pontus-osterdahl**
This pull request adds the functionality to edit and delete info-comments. Fixes #5702 

* Add editCommentDialog.xhtml

* Enable removal of Comment in CommentService and CommentsForm

* Add integration test for CommentService

* Changed properties of comments association in Process bean

* Add messages for editing and deleting comments

* Update kitodo.css

### CODE REVIEW

File: Kitodo/src/main/java/org/kitodo/production/forms/CommentForm.java [hunk 1]
```diff
@@ -59,6 +60,19 @@ public Process getProcess() {
...
+    /**
+     * Remove a given comment.
+     * 
+     * @param comment to be removed.
+     */
+    public void removeComment(Comment comment) {
+        try {
+            ServiceManager.getCommentService().removeComment(comment);
+        } catch (DAOException e) {
+            e.printStackTrace();
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/production/forms/CommentForm.java:72]: Use a right logging statement or use the helper method to get the error displayed to the user.

File: Kitodo/src/main/java/org/kitodo/production/services/data/CommentService.java [hunk 2]
```diff
@@ -79,4 +84,18 @@ public List<Comment> getAllCommentsByProcess(Process process) {
...
+     * Remove comment from database and resolve associations.
+     * 
+     * @param comment to be removed.
+     */
+    public static void removeComment(Comment comment) throws DAOException {
+        comment.getProcess().getComments().remove(comment);
+        comment.setProcess(null);
+        comment.setAuthor(null);
+        comment.setCorrectionTask(null);
+        comment.setCurrentTask(null);
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/production/services/data/CommentService.java:98]: Is this really necessary if the comment will get deleted?

**Review by henning-gerhardt — COMMENTED** [review]
fast code only review

**Comment by pontus-osterdahl** [comment]
Added changes suggested in comments to https://github.com/kitodo/kitodo-production/issues/5702 (only author is allowed to edit/delete comment and only most recent comment should be edited).

**Review by henning-gerhardt — COMMENTED** [review]
Changes looks good (except returning `null` but maybe needed in the CommentForm class usage in the UI) but I did not test them nor I'm familiar with the JSF and CSS code, so somebody else should have a more closer look into this and maybe did even test it.

**Review by oliver-stoehr — CHANGES_REQUESTED** [review]
When I edit a comment in the metadata editor, save the comment and then do one of the following actions:
- Delete the comment
- Click "Exit"
- Click "Save"
- Click "Save & exit"

I run into the following exception:
`javax.el.PropertyNotFoundException: Target Unreachable, [editedComment] returned null`
<details>
<summary>Stacktrace</summary>

```
javax.el.PropertyNotFoundException: Target Unreachable, [editedComment] returned null
	at org.apache.el.parser.AstValue.getTarget(AstValue.java:123)
	at org.apache.el.parser.AstValue.getType(AstValue.java:57)
	at org.apache.el.ValueExpressionImpl.getType(ValueExpressionImpl.java:173)
	at org.jboss.weld.el.WeldValueExpression.getType(WeldValueExpression.java:93)
	at org.jboss.weld.el.WeldValueExpression.getType(WeldValueExpression.java:93)
	at org.apache.myfaces.view.facelets.el.ContextAwareTagValueExpression.getType(ContextAwareTagValueExpression.java:76)
	at org.primefaces.util.ComponentUtils.getConverter(ComponentUtils.java:160)
	at org.primefaces.renderkit.InputRenderer.getConvertedValue(InputRenderer.java:45)
	at javax.faces.component.UIInput.getConvertedValue(UIInput.java:789)
	at javax.faces.component.UIInput.validate(UIInput.java:708)
	at javax.faces.component.UIInput.processValidators(UIInput.java:293)
	at org.apache.myfaces.context.servlet.PartialViewContextImpl$PhaseAwareVisitCallback.visit(PartialViewContextImpl.java:779)
	at org.apache.myfaces.component.visit.PartialVisitContext.invokeVisitCallback(PartialVisitContext.java:213)
	at javax.faces.component.UIComponent.visitTree(UIComponent.java:1019)
	at javax.faces.component.UIComponentBase.visitTree(UIComponentBase.java:1192)
	at javax.faces.component.UIForm.visitTree(UIForm.java:357)
	at javax.faces.component.UIComponent.visitTree(UIComponent.java:1047)
	at javax.faces.component.UIComponentBase.visitTree(UIComponentBase.java:1192)
	at javax.faces.component.UIComponent.visitTree(UIComponent.java:1047)
	at javax.faces.component.UIComponentBase.visitTree(UIComponentBase.java:1192)
	at javax.faces.component.UIComponent.visitTree(UIComponent.java:1047)
	at javax.faces.component.UIComponentBase.visitTree(UIComponentBase.java:1192)
	at javax.faces.component.UIComponent.visitTree(UIComponent.java:1047)
	at javax.faces.component.UIComponentBase.visitTree(UIComponentBase.java:1192)
	at org.apache.myfaces.context.servlet.PartialViewContextImpl.processPartialExecute(PartialViewContextImpl.java:435)
	at org.apache.myfaces.context.servlet.PartialViewContextImpl.processPartial(PartialViewContextImpl.java:416)
	at org.primefaces.context.PrimePartialViewContext.processPartial(PrimePartialViewContext.java:65)
	at javax.faces.context.PartialViewContextWrapper.processPartial(PartialViewContextWrapper.java:97)
	at org.omnifaces.context.OmniPartialViewContext.processPartial(OmniPartialViewContext.java:122)
	at javax.faces.component.UIViewRoot$ProcessValidatorPhaseProcessor.process(UIViewRoot.java:1880)
	at javax.faces.component.UIViewRoot._process(UIViewRoot.java:1738)
	at javax.faces.component.UIViewRoot.processValidators(UIViewRoot.java:982)
	at org.apache.myfaces.lifecycle.ProcessValidationsExecutor.execute(ProcessValidationsExecutor.java:38)
	at org.apache.myfaces.lifecycle.LifecycleImpl.executePhase(LifecycleImpl.java:195)
	at org.apache.myfaces.lifecycle.LifecycleImpl.execute(LifecycleImpl.java:142)
	at javax.faces.webapp.FacesServlet.service(FacesServlet.java:204)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:209)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.kitodo.production.servletfilter.EncodingFilter.doFilter(EncodingFilter.java:68)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.omnifaces.filter.GzipResponseFilter.doFilter(GzipResponseFilter.java:183)
	at org.omnifaces.filter.HttpFilter.doFilter(HttpFilter.java:108)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:337)
	at org.kitodo.production.security.SecurityObjectAccessFilter.doFilter(SecurityObjectAccessFilter.java:89)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:115)
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:81)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:122)
	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:116)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:126)
	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:81)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
	at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:109)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
	at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:149)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
	at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
	at org.springframework.security.web.session.ConcurrentSessionFilter.doFilter(ConcurrentSessionFilter.java:147)
	at org.springframework.security.web.session.ConcurrentSessionFilter.doFilter(ConcurrentSessionFilter.java:125)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:223)
	at org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter.doFilter(AbstractAuthenticationProcessingFilter.java:217)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:103)
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:89)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
	at org.springframework.security.web.header.HeaderWriterFilter.doHeadersAfter(HeaderWriterFilter.java:90)
	at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:75)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:112)
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:82)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
	at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:55)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
	at org.springframework.security.web.session.DisableEncodeUrlFilter.doFilterInternal(DisableEncodeUrlFilter.java:42)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:346)
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:221)
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:186)
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:354)
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:267)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.apache.logging.log4j.web.Log4jServletFilter.doFilter(Log4jServletFilter.java:71)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:167)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:481)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:130)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:673)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:390)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:926)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1791)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:829)
```
</details>

**Comment by solth** [comment]
@pontus-osterdahl could you take a look at the problem reported by @oliver-stoehr ?

**Comment by pontus-osterdahl** [comment]
@oliver-stoehr Thank you for your review and comment!
Unfortunately, I am not able to reproduce the exception.

I first edited and saved an existing comment in the metadata editor and then performed the mentioned actions, but no exception occurs. 

Have I misunderstood your description of the problem?

**Comment by oliver-stoehr** [comment]
I think you got it right, but I recorded a video just to be clear:
error

It seems like the `editCommentDialog.xhtml` is rendered after an action like pressing "Save" updated the page. I think the rendering fails because `editedComment` is already set to `null` at that point.
Adding `rendered="#{CommentForm.getEditedComment() ne null}"` to the `editCommentForm` seems to fix the issue!

**Review by oliver-stoehr — APPROVED** [review]

**Comment by solth** [comment]
@pontus-osterdahl would you mind rebasing your branch against the current master since there have been some fixes for the tests and builds since you first opened this pull request?

**Comment by pontus-osterdahl** [comment]
@solth done.

**Comment by solth** [comment]
@pontus-osterdahl thank you for the update. I think something went wrong with the Github logic, though, since your pull request now lists 113 changed files with 2052 added and 474 deleted lines in those files:

Bildschirmfoto 2023-09-21 um 14 07 57

Your pull request now seems to (wrongly) list all files changed in the master since you first opened your pull request as changed of your pull request and differences between your branch and the master.

The commit message of the merge commit also looks a little suspicious:

Bildschirmfoto 2023-09-21 um 14 14 52

Shouldn't this read "Merge branch 'master' of https://github.com/kitodo/kitodo-production into Enable-editing-and-delete-info-comments"? 

Clicking on your branch in your fork however correctly states the number of changed files and lines:

Bildschirmfoto 2023-09-21 um 14 09 38

so I guess the branch itself is in the correct state, just the pull request is somehow broken.

I am not sure what went wrong here but I have a bad feeling about merging this pull request in this form. Could you open a new pull request from the branch - without any additional changes - to see if it will state the number of changes correctly?

**Comment by pontus-osterdahl** [comment]
@solth Thanks for your comment and please excuse the inconvenience! I opened a new pull request #5785.

**Pull request merged**